### DISCUSSION TITLE
Incorrect Dependency Linking in Dependency Graph on GitHub

### DISCUSSION THREAD

**Discussion by ZjzMisaka** [community/community#63205] [Packages]
### Select Topic Area

Question

### Body

I am encountering an issue with the Dependency Graph feature in GitHub. I have two projects, DataTransformer and RoslynScriptRunner, both hosted on GitHub. The DataTransformer project has a dependency on the RoslynScriptRunner NuGet package, which is also my own project.

The problem is that in the Dependency Graph of the DataTransformer project, the RoslynScriptRunner dependency is not correctly linked to its corresponding repository. Additionally, in the RoslynScriptRunner repository, the Dependents list does not show the DataTransformer project, even though it uses RoslynScriptRunner as a dependency.

I've tried renaming the RoslynScriptRunner repository to ZjzMisaka.RoslynScriptRunner to match the NuGet package name (Since NuGet packages do not allow 'Roslyn' as a prefix, I added my own ID as a prefix and published it to NuGet as 'ZjzMisaka.RoslynScriptRunner'), but this did not resolve the issue.

I would appreciate any assistance or guidance on how to correct this so that the Dependency Graph correctly reflects and links the dependencies.

**Comment by nirav-gajera** [comment]
The Dependency Graph feature in GitHub uses the following information to determine the dependencies of a project:

* The manifest files of the project, such as the `package.json` file for JavaScript projects and the `pom.xml` file for Java projects.
* The NuGet packages that are installed in the project.
* The GitHub repositories that are linked to the project.

In your case, the Dependency Graph is not correctly reflecting the dependencies of the DataTransformer project because the RoslynScriptRunner NuGet package is not linked to the correct repository. This is likely because the name of the RoslynScriptRunner repository does not match the name of the NuGet package.

To fix this issue, you need to rename the RoslynScriptRunner repository to match the name of the NuGet package. You can do this by going to the "Settings" page for the RoslynScriptRunner repository and clicking on the "Repositories" tab. Then, click on the "Rename" button and enter the new name for the repository.

Once you have renamed the RoslynScriptRunner repository, you need to wait for GitHub to update the Dependency Graph. This may take a few minutes.

Once the Dependency Graph has been updated, the RoslynScriptRunner dependency should be correctly linked to its corresponding repository. Additionally, the Dependents list for the RoslynScriptRunner repository should show the DataTransformer project.

If you are still having problems after following these steps, you can contact GitHub Support for assistance.

**Reply by ZjzMisaka** [reply]
Thank you, but I waited for whole day after changing the project name and the link still didn't work😅. maybe I should try contact GitHub Support for assistance.
